apps/sms/views/conversation/js/compose.js

```diff
@@ -487,25 +469,16 @@ var Compose = (function() {
       return state.empty;
     },
 
-    /**
-     * Lock composer when size limit is reached.
-     */
-    lock: function() {
-      state.locked = true;
-      dom.attachButton.disabled = true;
-    },
-
-    /**
-     * Unlock composer when size is decreased again.
-     */
-    unlock: function() {
-      state.locked = false;
-      dom.attachButton.disabled = false;
+    setupLock({ canSend, canEdit, getType }) {
```

Comment:
I didn't find better way to clean ```canSend```, ```canEdit``` and ```getType``` from the previous panel, I tried to clean it in ```clear``` but it's called from a lot of places and sometimes asynchronously (e.g. draft load case) and it may unexpectedly clean our "lock", so I just always call setup lock in beforeEnter. Also we won't need this "workaround" in nga.